subprojects/core/src/main/java/org/gradle/internal/operations/trace/SerializedOperationFinish.java

```diff
@@ -37,11 +39,27 @@
     SerializedOperationFinish(BuildOperationDescriptor descriptor, OperationFinishEvent finishEvent) {
         this.id = ((OperationIdentifier) descriptor.getId()).getId();
         this.endTime = finishEvent.getEndTime();
-        this.result = finishEvent.getResult();
+        this.result = transform(finishEvent.getResult());
         this.resultClassName = result == null ? null : finishEvent.getResult().getClass().getName();
         this.failureMsg = finishEvent.getFailure() == null ? null : finishEvent.getFailure().toString();
     }
 
+    private Object transform(Object details) {
```

Comment:
If we want to keep going this path (serializing the details), we should make the serializer/deserializer pluggable and put it in the right subproject. I'm not sure why we try to serialize all details objects though. @ldaley what is your take on this?

Reply:
we currently only use it for our `-Dorg.gradle.internal.operations.trace` flag to get the build operations to disk. This is also used in the `BuildOperationTestFixture`

Reply:
@oehme as Rene says, this is just for testing and debugging. 

I think we are close to the threshold of needing a better solution, but we aren't there yet.